javaslang/src/main/java/javaslang/collection/BitMappedTrie.java

```diff
@@ -264,8 +255,6 @@ T get(int index) {
     Object getLeaf(int index) {
         if (depthShift == 0) {
             return array;
-        } else if (depthShift == BRANCHING_BASE) {
```

Comment:
this separate optimization didn't have a big advantage, as the general case already covers it.